solana-programs/staking-bridge/tests/wormholeTestUtils.ts

```diff
@@ -0,0 +1,22 @@
+import * as anchor from '@coral-xyz/anchor'
+import { SOL_AUDIO_DECIMALS, SOL_USDC_DECIMALS } from './constants';
+
+// Remove the 0x prefix and pad to 32 bytes
+export function formatEthAddress(address: string) {
+  return address.slice(2).padStart(64, '0')
+}
+
+export function getPostMessageData(args: { wholeAmount: number, solTokenDecimals: number, lastValidBlockHeight: number }) {
+  return {
+    nonce: new anchor.BN(args.lastValidBlockHeight),
+    amount: new anchor.BN(args.wholeAmount * 10 ** args.solTokenDecimals)
+  }
+}
+
+export function getMinimumAmountOutFromAmountIn(wholeAmountIn: number) {
```

Comment:
Ok just so I understand, this method is something you pass into raydium where you tell it at least how much you expect to come out to prevent crazy slippage.

Since this is a permissionless program, I think we should set this value to zero. There is nothing we can do about this. but it is interesting to think about.

Reply:
your understanding is correct, but i don't understand why we should set to 0. what if slippage happens to be super high and we lose most of our USDC? wouldn't it be better to fail and wait for later?